test/e2e/api/events/map-change-event.test.js

```diff
@@ -0,0 +1,57 @@
+import { test, expect, chromium } from '@playwright/test';
+
+test.describe('map change event test ', () => {
+  let page;
+  let context;
+  test.beforeAll(async () => {
+    context = await chromium.launchPersistentContext('');
+    page =
+      context.pages().find((page) => page.url() === 'about:blank') ||
+      (await context.newPage());
+    await page.goto('events/map-change-event.html');
+  });
+
+  test.afterAll(async function () {
+    await context.close();
+  });
+
+  test('Map change event for layers work', async () => {
+    let layerClicked = 0;
+    page.on('console', (msg) => {
+      if (msg.text() === 'Layer clicked') {
+        layerClicked++;
+      }
+    });
+
+    await page.evaluate(() => {
+      const layer = document.querySelector('map-layer');
+      layer.addEventListener('map-change', () => {
+        console.log('Layer clicked');
+      });
+      layer.checked = false;
+      layer.checked = true;
```

Comment:
may be good to test that the 'map-change' event is called in all possible cases, i.e. by clicking on the checkbox, using the 'checked' property or using [setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute).